### PR TITLE
Add prerequisite check

### PR DESCRIPTION
**Pull request by kezhenxu94**

### CODE REVIEW

**Review by wu-sheng — APPROVED** [review]

File: deploy/platform/kubernetes/Makefile [hunk 1]
```diff
@@ -155,9 +156,10 @@ feature-mesh-with-agent:
 deploy: $(features)
 	helm dep up .
 	helm -n $(NAMESPACE) upgrade --install $(RELEASE) . --create-namespace --timeout=20m $(HELM_OPTIONS)
-
+ifneq ($(DEBUG),true)
```
Comment (innerpeacez) [inline deploy/platform/kubernetes/Makefile:159]: Does only Istio-related sidecar automatically injected?

Comment (kezhenxu94) [reply deploy/platform/kubernetes/Makefile:159]: What do you mean? This is nothing related to Istio

Comment (innerpeacez) [reply deploy/platform/kubernetes/Makefile:159]: I'm curious about the purpose of this step (rollout restart).

Comment (innerpeacez) [reply deploy/platform/kubernetes/Makefile:159]: make sense

**Pull request merged**